firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -546,7 +550,9 @@ class SearchViewController: SiteTableViewController,
     }
 
     func loader(dataLoaded data: Cursor<Site>) {
-        self.data = data
+        self.data = (profile.prefs.boolForKey(PrefsKeys.FirefoxSuggestShowSponsoredSuggestions) ?? false)
```

Comment:
Would you mind breaking this up into an `if`/`else`, please? I think the combination of `??` and `?:` make this expression a little tough to read.

The logic looks good, though; thanks! 🎉 I think, at one time, `Cursor` might've been used for lazy loading? But it looks like `RustPlaces` also uses an `ArrayCursor` under the hood now, and I bet it's faster to do it your way than to get fancy by chaining `filter` and `compactMap`.

Reply:
i was debating between an if/else or a ternary, they both looked kinda clunky

Reply:
I hear you! Swift 5.9 added `if` expressions, so maybe that (plus lifting the pref check into a variable) might help a little?

```swift
let showSponsoredSuggestions = profile.prefs.boolForKey(PrefsKeys.FirefoxSuggestShowSponsoredSuggestions) ?? false
self.data = if showSponsoredSuggestions {
    ArrayCursor<Site>(data: SponsoredContentFilterUtility().filterSponsoredSites(from: data.asArray()))
} else {
    data
}
```

Still a little wordy, but gives the operators a bit more room to breathe. But I like the way you have it now, too; no need to change it anymore!